// Copyright 2021 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wfa.panelmatch.protocol;

option java_package = "org.wfanet.panelmatch.protocol";
option java_multiple_files = true;

// The request to encrypt.
message CryptorEncryptRequest {
  repeated bytes plaintexts = 1;
  bytes encryption_key = 2;
}

// Response of the CryptorEncryptRequest method.
message CryptorEncryptResponse {
  repeated bytes encrypted_texts = 1;
  int64 elapsed_cpu_time_millis = 2;
}

// The request to reencrypt.
message CryptorReEncryptRequest {
  repeated bytes encrypted_texts = 1;
  bytes encryption_key = 2;
}

// Response of the CryptorReEncryptRequest method.
message CryptorReEncryptResponse {
  repeated bytes reencrypted_texts = 1;
  int64 elapsed_cpu_time_millis = 2;
}

// The request to decrypt.
message CryptorDecryptRequest {
  repeated bytes encrypted_texts = 1;
  bytes encryption_key = 2;
}

// Response of the CryptorEncryptRequest method.
message CryptorDecryptResponse {
  repeated bytes decrypted_texts = 1;
  int64 elapsed_cpu_time_millis = 2;
}
